learning_paths/api/v1/views.py

```diff
@@ -105,3 +118,233 @@ def get(self, request, learning_path_uuid):
+        )
+        if username:
+            enrollment = enrollment.filter(user__username=username).first()
+        else:
+            enrollment = enrollment.filter(user=request.user).first()
+
+        if not enrollment:
+            return Response(
+                {"detail": "No enrollment."}, status=status.HTTP_404_NOT_FOUND
+            )
+
+        if (
+            not request.user.is_staff
+            and not request.user.is_superuser
+            and request.user != enrollment.user
+        ):
+            raise PermissionDenied
+
+        if (
+            not settings.LEARNING_PATHS_ALLOW_SELF_UNENROLLMENT
+            and request.user == enrollment.user
+        ):
+            raise PermissionDenied
+
+        enrollment.is_active = False
+        enrollment.save()
+        return Response(
+            LearningPathEnrollmentSerializer(enrollment).data,
+            status=status.HTTP_204_NO_CONTENT,
+        )
+
+
+class FetchEnrollmentsView(APIView):
+    """
+    A view to fetch learning path enrollments for the authenticated user.
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        """
+        Handle GET requests to fetch enrollments.
+        """
+        if request.user.is_staff or request.user.is_superuser:
+            enrollments = LearningPathEnrollment.objects.all()
+        else:
+            enrollments = LearningPathEnrollment.objects.filter(user=request.user)
+
+        return Response(LearningPathEnrollmentSerializer(enrollments, many=True).data)
+
+
+class BulkEnrollAPIView(APIView):
+    """
+    Bulk enrollment API for LearningPathEnrollment.
+
+    For existing users, it creates a new LearningPathEnrollment record.
+    For non-existing users, it creates a new LearningPathEnrollmentAllowed record.
+    """
+
+    permission_classes = [IsAdminUser]
+
+    def post(self, request, *args, **kwargs):
+        """
+        POST requests handler for the view.
+        """
+        data = request.data
+        learning_paths_keys = data.get("learning_paths", "").split(",")
+        emails = data.get("emails", "").split(",")
+
+        valid_learning_paths_keys = []
+        for key in learning_paths_keys:
+            try:
+                UUID(key)
+            except ValueError:
+                logger.warning("BulkEnrollAPIView: Invalid learning path key: %s", key)
+                continue
+            valid_learning_paths_keys.append(key)
+
+        learning_paths = LearningPath.objects.filter(
+            uuid__in=valid_learning_paths_keys
+        ).all()
+
+        existing_users = User.objects.filter(email__in=emails).all()
+        non_existing_emails = set(emails) - set(u.email for u in existing_users)
+
+        enrollments_created = []
+        enrollment_allowed_created = []
+
+        for learning_path in learning_paths:
+
+            # Create LearningPathEnrollment, LearningPathEnrollmentAllowed for existing users
```

Comment:
This one does not match the docstring of this class. Why do we create `LearningPathEnrollmentAllowed` for the existing users?

Reply:
So, this is something I put it in just in case and then rework. The purpose is to store some "Audit" information. I was expected that this would store a few more fields than just email, so I had the object created in case more information needed to be stored later.

I am not super clear on the audit requirements. We can remove this if this metdata doesn't have to be associated with existing enrollments.

Reply:
@tecoholic, yes, let's remove it for now.